backend/app/training/scrape_products.py

```
@@ -251,18 +252,54 @@ async def scrape_products(sitemap_url, scrape_limit=float('inf'), sample_size=5,
+            print(f"Success with discovered sitemap: {first}")
+            return products
+        else:
+            print(f"No products found in discovered sitemap: {first}")
+
+    # 2) Fallback to user-provided sitemap
+    if sitemap_url:
+        products = await _attempt(sitemap_url)
+        if products:
+            print(f"Success with provided sitemap: {sitemap_url}")
+            return products
+        else:
```

Ruff (0.8.2)

Unnecessary `else` after `return` statement

Remove unnecessary `else`

(RET505)